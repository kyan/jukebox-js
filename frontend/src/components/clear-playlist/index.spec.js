import React from 'react'
import { shallow } from 'enzyme'
import { Confirm } from 'semantic-ui-react'
import ClearPlaylist from './index'

describe('ClearPlaylist', () => {
  const onClearMock = jest.fn()
  let wrapper

  describe('render', () => {
    wrapper = shallow(
      <ClearPlaylist
        onClear={onClearMock}
      />
    )

    it('renders the as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    describe('confirm dialog', () => {
      it('is not shown default', () => {
        expect(wrapper.find(Confirm).prop('open')).toEqual(false)
      })

      it('it shows when the button is pressed', () => {
        wrapper.find('.jb-clear-button').simulate('click')
        expect(wrapper.find(Confirm).prop('open')).toEqual(true)
      })
    })

    describe('callbacks', () => {
      it('calls the onConfirm callback', () => {
        wrapper.find('.jb-clear-button').simulate('click')
        expect(wrapper.instance().state.open).toEqual(true)
        wrapper.find(Confirm).prop('onConfirm')()
        expect(onClearMock.mock.calls.length).toEqual(1)
      })

      it('calls the onCancel callback', () => {
        wrapper.find('.jb-clear-button').simulate('click')
        wrapper.find(Confirm).prop('onCancel')()
        expect(wrapper.instance().state.open).toEqual(false)
      })
    })
  })
})
