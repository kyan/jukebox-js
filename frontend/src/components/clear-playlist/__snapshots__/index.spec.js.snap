// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearPlaylist render renders the as expected 1`] = `
<span>
  <Label
    as="a"
    className="jb-clear-button"
    color="red"
    horizontal={true}
    onClick={[Function]}
    size="mini"
  >
    CLEAR
  </Label>
  <Confirm
    cancelButton="No thanks"
    confirmButton="Do it!"
    content="Are you sure you want to nuke the playlist?"
    onCancel={[Function]}
    onConfirm={[Function]}
    open={false}
    size="small"
  />
</span>
`;
